components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/RoleManagementService.java

```diff
@@ -175,4 +175,17 @@ RoleBasicInfo setPermissionsForRole(String roleID, List<String> permissions, Str
      * @throws IdentityRoleManagementException IdentityRoleManagementException.
      */
     boolean isExistingRole(String roleID, String tenantDomain) throws IdentityRoleManagementException;
+
+    /**
+     * Check whether the given role name exist.
+     *
+     * @param roleName     Role name.
+     * @param tenantDomain Tenant domain.
+     * @return {@code true} if the the given role ID exist.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    default boolean isExistingRoleName(String roleName, String tenantDomain) throws IdentityRoleManagementException {
+
+        throw new IdentityRoleManagementException("isExistingRoleName method is not implemented");
```

Comment:
are we handling this exception properly? In case of this exception, we need to have the old behavior, which is to continue without validation

Reply:
Fixed: 3b8457ce1a80e96bdf0458aea59907543b13f5ba